src/EditorFeatures/Test2/FindReferences/FindReferencesTests.vb

```diff
@@ -586,5 +586,35 @@ partial class C
             Dim linkedSymbols = Await SymbolFinder.FindLinkedSymbolsAsync(symbol1, solution, cancellationToken:=Nothing)
             Assert.Equal(expectedLinkedSymbolCount, linkedSymbols.Length)
         End Function
+
+        <Fact, WorkItem(1758726, "https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1758726")>
+        Public Async Function TestFindReferencesInDocumentsNoCompilation() As Task
+            Using workspace = TestWorkspace.Create("
+<Workspace>
+    <Project Language=""NoCompilation"" AssemblyName=""NoCompilationAssembly"" CommonReferencesPortable=""true"">
+        <Document>
+            var x = {}; // e.g., TypeScript code or anything else that doesn't support compilations
+        </Document>
+    </Project>
+    <Project Language=""C#"" AssemblyName=""CSharpAssembly"" CommonReferencesPortable=""true"">
+        <Document>
+class C
+{
+}
+        </Document>
+    </Project>
+</Workspace>
+", composition:=s_composition)
+                Dim solution = workspace.CurrentSolution
+                Dim csProject = solution.Projects.Single(Function(p) p.SupportsCompilation)
+                Dim compilation = Await csProject.GetCompilationAsync()
+                Dim symbol = compilation.GetTypeByMetadataName("C")
+
+                Dim progress = New StreamingFindReferencesProgressAdapter(NoOpFindReferencesProgress.Instance)
+                Await SymbolFinder.FindReferencesInDocumentsInCurrentProcessAsync(
```

Comment:
what is this asserting?

Reply:
that we don't crash :)  without the fix this throws.